src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -474,9 +477,10 @@ export class EditDataGridPanel extends GridParentComponent {
 	private replaceLinebreaks(inputStr: string): string {
 		let newlineMatches = inputStr.match(/(\r\n|\n|\r)/g);
 		if (newlineMatches && newlineMatches.length > 0) {
-			this.newlinePattern = newlineMatches[0];
+			this.hasNewline = true
 		}
-		return inputStr.replace(/(\r\n|\n|\r)/g, '\u0000');
+		// allow-any-unicode-next-line
+		return inputStr.replace(/\r\n/g, '\u0000').replace(/\n/g, '↵').replace(/\r/g, '⏎');
```

Comment:
I don't think this is a good solution. I'd just keep the current behavior. if you really want to make it work, you should adopt the approach I mentioned earlier.
cons of this implementation:
1. there are more scenarios you have to raise the unsupported error, previously it is only the NULL value, now you are adding 2 more.
2. inconsistent, some line breaks are visible and some are not.